pkg/node/node.go

```diff
@@ -955,9 +956,15 @@ func NewBee(interrupt chan struct{}, sysInterrupt chan os.Signal, addr string, p
 		return nil, fmt.Errorf("pullsync protocol: %w", err)
 	}
 
+	staking := staking.New(overlayEthAddress, chainCfg.StakingContract, erc20Address, transactionService, common.BytesToHash(nonce))
+
 	if o.FullNodeMode {
 		depthMonitor := depthmonitor.New(kad, pullSyncProtocol, storer, batchStore, logger, warmupTime, depthmonitor.DefaultWakeupInterval)
 		b.depthMonitorCloser = depthMonitor
+
+		// TODO: inject incentives contract, blocksRound, blocksPerPhase
+		agent := storageincentives.New(swarmAddress, chainBackend, logger, depthMonitor, nil, batchStore, storer, o.BlockTime, storageincentives.DefaultBlocksPerRound, storageincentives.DefaultBlocksPerPhase)
```

Comment:
This PR is complete?

This agent will run only if we are staked right?

Also, this is sort of a chicken and egg problem. We allow user to stake using the API, but the agent is only started on startup. So if user stakes using API later he will need to restart the node right?

Also, this `IsStaked` function that you have implemented, how is this going to be used?

Reply:
@zelig stated the same problem. The agent runs on full-node if the node has staked or not. So this solves the chicken and egg problem.

Reply:
see https://github.com/ethersphere/bee/pull/3364#discussion_r984896894

Reply:
Should we use the IsStaked function in the agent worker? Or we dont need it?